common/src/main/java/io/druid/guice/DruidSecondaryModule.java

```diff
@@ -75,24 +76,36 @@ public void configure(Binder binder)
     binder.bind(JsonConfigurator.class).toInstance(jsonConfigurator);
   }
 
-  @Provides @LazySingleton @Json
+  @Provides
+  @LazySingleton
+  public InjectableValues getGuiceInjectableValues(final Injector injector)
+  {
+    return new GuiceInjectableValues(injector);
+  }
+
+  @Provides
+  @LazySingleton
+  @Json
   public ObjectMapper getJsonMapper(final Injector injector)
   {
     setupJackson(injector, jsonMapper);
     return jsonMapper;
   }
 
-  @Provides @LazySingleton @Smile
+  @Provides
+  @LazySingleton
+  @Smile
   public ObjectMapper getSmileMapper(Injector injector)
   {
     setupJackson(injector, smileMapper);
     return smileMapper;
   }
 
-  private void setupJackson(Injector injector, final ObjectMapper mapper) {
+  private void setupJackson(Injector injector, final ObjectMapper mapper)
+  {
     final GuiceAnnotationIntrospector guiceIntrospector = new GuiceAnnotationIntrospector();
 
-    mapper.setInjectableValues(new GuiceInjectableValues(injector));
+    mapper.setInjectableValues(injector.getInstance(InjectableValues.class));
```

Comment:
`GuiceInjectableValues` needs injected to QueryResource to allow the ObjectMapper copies to retain their InjectableValues. Since I was moving an instance of `GuiceInjectableValues` to Guice injection (confusing, I know) I could either have added the InjectableValue as a parameter to the static method or simply pulled it from the injector. I opted for the later to have the minimum overall code change to the file, but have no problems with doing it via passing parameters.